File: spring-rest-configuration-autodiscover.java/src/main/java/demo/repository/PropertyRepository.java
package demo.repository;

import demo.domain.Property;
import demo.service.core.Code;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface PropertyRepository extends JpaRepository<Property, Code> {

    Optional<Property> findByCode(Code code);
}

File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/repository/PropertyRepository.scala
package demo.repository

import java.util.Optional

import demo.domain.Property
import demo.service.core.Code
import org.springframework.data.jpa.repository.JpaRepository

trait PropertyRepository extends JpaRepository[Property, Code.Value] {

  def findByCode(code: Code.Value): Optional[Property]
}
